### DISCUSSION TITLE
Popups

### DISCUSSION THREAD

**Discussion by ghost** [slint-ui/slint#647] [Q&A]
How should I handle popup windows? They're normally very useful, but I don't currently see a way to handle them.

My first thought was to add a `Popup` widget and export it in my main `.60` file, but only my main `App` widget gets exported. If I declare the popup in the main file, they only it gets exported and now I can't run my app at all, just my popups.

**Comment by ogoffart** [comment]
What exactly do you mean by `Popup`
Is it something like a popup menu. Then you can use the `PopupWindow` element.  This is what the ComboBox widget uses.

The fact that one currently can't export several root component is a known problem. This is currently not yet implemented.

**Reply by ghost** [reply]
By `Popup` I mean a new window that has some options you can click.

**Reply by ogoffart** [reply]
So you mean a modal dialog then?

**Reply by ghost** [reply]
Yeah, a modal dialog

**Reply by ogoffart** [reply]
This is currently not implemented. But this is a feature we want to have.

**Reply by JuliDi** [reply]
Does this imply that it is not (yet) possible to have multiple windows (or popups/modal dialogs etc) in one application?

E.g. showing error messages in a modal dialog would seem pretty handy. The `Dialog` element seems like a good start, but I don't quite understand how to use it. I expected it to be usable like a modal dialog, i.e., you can make it pop up (which blocks the program flow at that point), the user selects "yes" or "no" and then it returns the selection to the program or so. Is that possible somehow?

**Reply by Xiaobaishushu25** [reply]
> Does this imply that it is not (yet) possible to have multiple windows (or popups/modal dialogs etc) in one application?
> 
> E.g. showing error messages in a modal dialog would seem pretty handy. The `Dialog` element seems like a good start, but I don't quite understand how to use it. I expected it to be usable like a modal dialog, i.e., you can make it pop up (which blocks the program flow at that point), the user selects "yes" or "no" and then it returns the selection to the program or so. Is that possible somehow?

i also have thie question，May I ask if it is currently possible to do so? I don't even know how to pop up a dialog.

**Reply by ogoffart** [reply]
It is actually possible to have separate window.
There is an example there: https://github.com/slint-ui/slint/pull/2094

The only thing is that they don't share the same globals and structures.

**Reply by Xiaobaishushu25** [reply]
> It is actually possible to have separate window.实际上可以有单独的窗口。 There is an example there: #2094那里有一个例子：#2094
> 
> The only thing is that they don't share the same globals and structures.

Thank you for you reply, this is a great example,i know how to pop up a dialog! but I think it's not very elegant to mix ui code and rust code together? In addition, i think the dialog should have some optional modality attributes, such as application_ Modal, when this window is displayed, other windows prohibit operations.
